cmake/FindJavaBuildTools.cmake

```diff
@@ -0,0 +1,26 @@
+find_program (GRADLE_EXECUTABLE gradle)
+find_program (MAVEN_EXECUTABLE mvn)
+
+set (GRADLE_FOUND FALSE)
+if (GRADLE_EXECUTABLE)
+    set (GRADLE_FOUND TRUE)
+    execute_process (
+        COMMAND ${GRADLE_EXECUTABLE} --version
+        OUTPUT_VARIABLE gradle_version)
+    message (STATUS "gradle found at ${GRADLE_EXECUTABLE}")
+    message (STATUS "${gradle_version}")
+else()
+    message (FATAL_ERROR "gradle was not found.")
+endif()
+
+set (MAVEN_FOUND FALSE)
+if (MAVEN_EXECUTABLE)
+    set (MAVEN_FOUND TRUE)
+    execute_process (
+        COMMAND ${MAVEN_EXECUTABLE} --version
+        OUTPUT_VARIABLE maven_version)
+    message (STATUS "maven found at ${MAVEN_EXECUTABLE}")
+    message (STATUS "${maven_version}")
```

Comment:
Will this output maven before the number version or just a random number?

Reply:
This will spit out a whole block of text. I couldn't find a option that gives a useful single line.
```
-- maven found at /usr/local/bin/mvn
-- Apache Maven 3.5.0 (ff8f5e7444045639af65f6095c62210b5713f426; 2017-04-03T19:39:06Z)
Maven home: /usr/local/maven-3.5.0
Java version: 1.8.0_144, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-8-oracle/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.4.0-93-generic", arch: "amd64", family: "unix"
```

gradle is similar:
```
-- gradle found at /usr/local/bin/gradle
-- 
------------------------------------------------------------
Gradle 4.0.1
------------------------------------------------------------
Build time:   2017-07-07 14:02:41 UTC
Revision:     38e5dc0f772daecca1d2681885d3d85414eb6826
Groovy:       2.4.11
Ant:          Apache Ant(TM) version 1.9.6 compiled on June 29 2015
JVM:          1.8.0_144 (Oracle Corporation 25.144-b01)
OS:           Linux 4.4.0-93-generic amd64
```

Reply:
ok this is fine